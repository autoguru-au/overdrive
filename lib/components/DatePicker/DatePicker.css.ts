import { style, styleVariants } from '@vanilla-extract/css';

import { cssLayerComponent } from '../../styles/layers.css';
import { overdriveTokens as vars } from '../../themes/theme.css';

export const inputContainer = style({
	'@layer': {
		[cssLayerComponent]: {
			display: 'grid',
			position: 'relative',
		},
	},
});

export const input = style({
	'@layer': {
		[cssLayerComponent]: {
			cursor: 'pointer',
			gridArea: '1 / 1',
			opacity: 0,
			position: 'relative',
			zIndex: 1,
			selectors: {
				'&::-webkit-calendar-picker-indicator': {
					background: 'transparent',
					color: 'transparent',
					cursor: 'pointer',
					height: '100%',
					left: 0,
					position: 'absolute',
					top: 0,
					width: '100%',
				},
			},
		},
	},
});

export const inputOverlay = style({
	'@layer': {
		[cssLayerComponent]: {
			gridArea: '1 / 1',
			pointerEvents: 'none',
			zIndex: 0,
		},
	},
});

export const contents = styleVariants({
	default: {
		'@layer': {
			[cssLayerComponent]: {
				alignItems: 'center',
				display: 'grid',
				gridGap: vars.space['1'],
				gridTemplateColumns: 'auto',
				justifyContent: 'flex-start',
			},
		},
	},
	withLabel: {
		'@layer': {
			[cssLayerComponent]: {
				gridTemplateColumns: 'auto auto',
			},
		},
	},
});

export const disabled = styleVariants({
	cursor: {
		'@layer': {
			[cssLayerComponent]: {
				cursor: 'not-allowed',
			},
		},
	},
	native: {
		'@layer': {
			[cssLayerComponent]: {
				opacity: '0.3',
			},
		},
	},
});
