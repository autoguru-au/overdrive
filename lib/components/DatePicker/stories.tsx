import {
	AccountEditIcon,
	AlertCircleIcon,
	CalendarIcon,
	CarIcon,
	CarMultipleIcon,
	CheckIcon,
	CurrencyUsdIcon,
	MagnifyIcon,
	PlusIcon,
	StarIcon,
} from '@autoguru/icons';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import * as React from 'react';
import { ComponentProps } from 'react';

import { Box } from '../Box';

import { DatePicker } from '.';

const iconOptions = {
	MagnifyIcon,
	CarIcon,
	CarMultipleIcon,
	CalendarIcon,
	AccountEditIcon,
	AlertCircleIcon,
	CurrencyUsdIcon,
	PlusIcon,
	StarIcon,
	CheckIcon,
};

export default {
	title: 'Components/DatePicker',
	component: DatePicker,
	decorators: [],
	argTypes: {
		icon: {
			defaultValue: void 0,
			description: 'Input field Icon',
			options: iconOptions,
			control: {
				type: 'select',
			},
		},
		size: {
			options: ['small', 'medium', 'large'],
			control: {
				type: 'select',
			},
		},
	},
} as ComponentMeta<typeof DatePicker>;

const Template: ComponentStory<typeof DatePicker> = (args) => (
	<Box display="flex">
		<DatePicker {...args} />
	</Box>
);

const standardProps: ComponentProps<typeof DatePicker> = {
	isLoading: false,
	disabled: false,
};

export const standard = Template.bind(standardProps);
standard.args = standardProps;

const smallProps: ComponentProps<typeof DatePicker> = {
	size: 'small',
	isLoading: false,
	disabled: false,
};

export const small = Template.bind(smallProps);
small.args = smallProps;

const smallWithLabelProps: ComponentProps<typeof DatePicker> = {
	...smallProps,
	valueLabel: 'Today',
};

export const smallWithLabel = Template.bind(smallWithLabelProps);
smallWithLabel.args = smallWithLabelProps;

const mediumProps: ComponentProps<typeof DatePicker> = {
	size: 'medium',
	isLoading: false,
	disabled: false,
};

export const medium = Template.bind(mediumProps);
medium.args = mediumProps;

const mediumWithLabelProps: ComponentProps<typeof DatePicker> = {
	...mediumProps,
	valueLabel: 'Today',
};

export const mediumWithLabel = Template.bind(mediumWithLabelProps);
mediumWithLabel.args = mediumWithLabelProps;

const largeProps: ComponentProps<typeof DatePicker> = {
	size: 'large',
	isLoading: false,
	disabled: false,
};

export const large = Template.bind(largeProps);
large.args = largeProps;

const largeWithLabelProps: ComponentProps<typeof DatePicker> = {
	...largeProps,
	valueLabel: 'Today',
};

export const largeWithLabel = Template.bind(largeWithLabelProps);
largeWithLabel.args = largeWithLabelProps;
