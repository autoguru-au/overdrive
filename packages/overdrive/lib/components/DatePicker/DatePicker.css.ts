import { style, styleVariants } from '@vanilla-extract/css';

import { themeContractVars as vars } from '../../themes/theme.css';

export const input = style({
	top: 0,
	right: 0,
	bottom: 0,
	left: 0,
	opacity: 0,
	cursor: 'pointer',
	selectors: {
		'&::-webkit-calendar-picker-indicator': {
			background: 'transparent',
			bottom: 0,
			color: 'transparent',
			cursor: 'pointer',
			height: 'auto',
			left: 0,
			position: 'absolute',
			right: 0,
			top: 0,
			width: 'auto',
		},
	},
});


export const contents = styleVariants({
	default: {
		display: 'grid',
		gridTemplateColumns: 'auto',
		alignItems: 'center',
		justifyContent: 'flex-start',
		gridGap: vars.space['1'],
	},
	withLabel: {
		gridTemplateColumns: 'auto auto',
	},
});
export const disabled = styleVariants({
	default: {
		cursor: 'not-allowed',
	},
	root: {
		opacity: '0.3',
	},
});

export const spinner = style({
	top: '50%',
	left: '50%',
	transform: 'translate(-50%, -50%)',
});
