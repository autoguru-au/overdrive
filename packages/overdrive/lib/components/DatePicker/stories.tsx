import {
	AccountEditIcon,
	AlertCircleIcon,
	CalendarIcon,
	CarIcon,
	CarMultipleIcon,
	CheckIcon,
	CurrencyUsdIcon,
	MagnifyIcon,
	PlusIcon,
	StarIcon,
} from '@autoguru/icons';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import * as React from 'react';
import { ComponentProps } from 'react';

import { DatePicker } from '.';

const iconOptions = {
	MagnifyIcon,
	CarIcon,
	CarMultipleIcon,
	CalendarIcon,
	AccountEditIcon,
	AlertCircleIcon,
	CurrencyUsdIcon,
	PlusIcon,
	StarIcon,
	CheckIcon,
};

export default {
	title: 'Components/DatePicker',
	component: DatePicker,
	decorators: [],
	argTypes: {
		icon: {
			defaultValue: void 0,
			description: 'Input field Icon',
			options: iconOptions,
			control: {
				type: 'select',
			},
		},
		size: {
			options: ['small', 'medium', 'large'],
			control: {
				type: 'select',
			},
		},
	},
} as ComponentMeta<typeof DatePicker>;

const Template: ComponentStory<typeof DatePicker> = (args) => (
	<DatePicker {...args} />
);

const standardProps: ComponentProps<typeof DatePicker> = {};

export const standard = Template.bind(standardProps);
standard.args = standardProps;
