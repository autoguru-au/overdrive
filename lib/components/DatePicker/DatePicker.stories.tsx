import type { Meta, StoryObj } from '@storybook/react-vite';
import React from 'react';

import { argTypesExampleIcons } from '../../stories/shared/argTypes';
import { Box } from '../Box/Box';

import { DatePicker } from './DatePicker';

const meta = {
	title: 'Components/Date Picker',
	component: DatePicker,
	decorators: [],
	argTypes: {
		icon: {
			defaultValue: null,
			description: 'Input field Icon',
			...argTypesExampleIcons,
		},
		size: {
			options: ['small', 'medium', 'large'],
			control: {
				type: 'select',
			},
		},
	},
} satisfies Meta<typeof DatePicker>;

export default meta;
type Story = StoryObj<typeof DatePicker>;

export const Standard: Story = {
	args: {
		isLoading: false,
		disabled: false,
	},
	render: (args) => (
		<Box display="flex">
			<DatePicker {...args} />
		</Box>
	),
};

export const SmallWithLabel: Story = {
	args: {
		size: 'small',
		isLoading: false,
		disabled: false,
		valueLabel: 'Today',
	},
	render: (args) => (
		<Box display="flex">
			<DatePicker {...args} />
		</Box>
	),
};

export const MediumWithLabel: Story = {
	args: {
		size: 'medium',
		isLoading: false,
		disabled: false,
		valueLabel: 'Today',
	},
	render: (args) => (
		<Box display="flex">
			<DatePicker {...args} />
		</Box>
	),
};

export const LargeWithLabel: Story = {
	args: {
		size: 'large',
		isLoading: false,
		disabled: false,
		valueLabel: 'Today',
	},
	render: (args) => (
		<Box display="flex">
			<DatePicker {...args} />
		</Box>
	),
};
