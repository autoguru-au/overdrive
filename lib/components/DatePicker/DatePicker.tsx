import { CalendarIcon, IconType } from '@autoguru/icons';
import {
	getLocalTimeZone,
	parseDate,
	today,
	type DateValue,
} from '@internationalized/date';
import clsx from 'clsx';
import React, {
	type ChangeEvent,
	useCallback,
	useEffect,
	useMemo,
	useState,
	forwardRef,
} from 'react';
import type { AriaPopoverProps } from 'react-aria';

import { elementStyles } from '../../styles/elementStyles';
import { sprinkles } from '../../styles/sprinkles.css';
import type { TestIdProp } from '../../types';
import { Calendar, type CalendarProps } from '../Calendar';
import { Icon } from '../Icon/Icon';
import { PopoverTrigger } from '../Popover';
import { ProgressSpinner } from '../ProgressSpinner/ProgressSpinner';
import { Text, type TextProps } from '../Text/Text';
import { VisuallyHidden } from '../VisuallyHidden/VisuallyHidden';

import * as styles from './DatePicker.css';

type SizeScale = 'small' | 'medium' | 'large';

const defaultEnglish = {
	openCalendar: 'open calendar',
} as const;

type LanguageEntries = Partial<Record<keyof typeof defaultEnglish, string>>;

export interface DatePickerProps extends TestIdProp {
	/**
	 * Options to customise the calendar: `allowPastDate`, `lang`, etc.
	 */
	calendarOptions?: Omit<CalendarProps, 'onChange'>;
	className?: string;
	/**
	 * Default selected date as an ISO string YYYY-MM-DD (uncontrolled)
	 */
	defaultValue?: string;
	/**
	 * Whether the picker is disabled and non-interactive
	 */
	disabled?: boolean;
	/**
	 * Icon to render inside the picker (defaults to calendar icon)
	 */
	icon?: IconType;
	/**
	 * Form field id
	 */
	id?: string;
	/**
	 * Show a loading state spinner instead of the icon
	 */
	isLoading?: boolean;
	/**
	 * Maximum selectable date YYYY-MM-DD
	 */
	max?: string;
	/**
	 * Minimum selectable date YYYY-MM-DD
	 */
	min?: string;
	/**
	 * Input field name recommended for form usage
	 */
	name?: string;
	onChange(date: string): void;
	/**
	 * Calendar popover placement relative to the trigger ('bottom left', 'top', etc.)
	 */
	placement?: AriaPopoverProps['placement'];
	/**
	 * Visual size of the picker control (small, medium, large)
	 */
	size?: SizeScale;
	/**
	 * The selected date as an ISO string (YYYY-MM-DD). Use `undefined` for an empty value
	 * (controlled)
	 */
	value?: string;
	/**
	 * Fallback label to display when no date value is selected.
	 */
	valueLabel?: string;
	/**
	 * Use native browser date picker instead of Calendar popover
	 */
	useNativePicker?: boolean;
	/**
	 * Language content override for DatePicker
	 */
	lang?: LanguageEntries;
}

const textSizeMap: Record<SizeScale, TextProps['size']> = {
	small: '2',
	medium: '3',
	large: '5',
};

const formatDateToString = (date: DateValue | null): string => {
	if (!date) return '';
	return date.toString(); // ISO format: YYYY-MM-DD to match native input
};

const parseDateString = (dateString: string): DateValue | null => {
	try {
		return parseDate(dateString);
	} catch {
		return null;
	}
};

/**
 * The DatePicker has been updated to render the Calendar component with a Popover while
 * maintaining backwards compatability.
 *
 * ## Props
 * - Visual size controlled by `size` (small | medium | large)
 * - Icon can be customized via the `icon` prop
 *
 * ## Event handling
 * - `onChange` is always invoked with the raw ISO date string (or empty string when cleared)
 *
 * ## Compatibility mode
 * - Setting `useNativePicker={true}` preserves the original browser-specific experience.
 *
 * ## Internationalization
 * - Override text values via `lang={{ openCalendar: 'open calendar' }}`
 * - Calendar options including `lang` can be passed via `calendarOptions` prop
 * - Date formatting helper available in `...utils/dateFormat.ts` or use `@internationalized/date` utils
 * - Advanced i18n and localization handled by [React Aria I18Provider](https://react-spectrum.adobe.com/react-aria/I18nProvider.html)
 * - Read more about [International calendars](https://react-spectrum.adobe.com/react-aria/useDatePicker.html#international-calendars)
 *
 * @example
 * // Uncontrolled component with default value
 * <DatePicker
 *   name="eventDate"
 *   defaultValue="2025-03-15"
 *   onChange={(isoDate) => console.log('Selected date:', isoDate)}
 * />
 *
 * @example
 * // Controlled component
 * <DatePicker
 *   name="bookingDate"
 *   value={selectedDate}
 *   valueLabel="Select a date"
 *   onChange={(isoDate) => setSelectedDate(isoDate)}
 *   calendarOptions={{
 *     allowPastDate: false,
 *     weekdayFormat: 'short',
 *   }}
 * />
 *
 * @example
 * // Native date picker usage
 * <DatePicker
 *   name="startDate"
 *   useNativePicker
 *   defaultValue="2025-03-12"
 *   onChange={(isoDate) => console.log('Native picked date:', isoDate)}
 * />
 */
export const DatePicker = forwardRef<HTMLInputElement, DatePickerProps>(
	(
		{
			calendarOptions,
			className,
			defaultValue,
			disabled = false,
			icon = CalendarIcon,
			isLoading = false,
			lang,
			max,
			min,
			onChange,
			placement = 'bottom left',
			size = 'medium',
			testId,
			useNativePicker = false,
			value,
			valueLabel,

			...inputProps
		},
		ref,
	) => {
		const isControlled = value !== undefined;

		const [selectedDate, setSelectedDate] = useState<DateValue | null>(
			() => {
				const initialValue = isControlled ? value : defaultValue;
				return initialValue ? parseDateString(initialValue) : null;
			},
		);
		const [popoverState, setPopoverState] = useState<{
			close: () => void;
		} | null>(null);

		// Sync external value changes (only for controlled components)
		useEffect(() => {
			if (isControlled) {
				const parsedDate = value ? parseDateString(value) : null;
				setSelectedDate(parsedDate);
			}
		}, [value, isControlled]);

		const onChangeEvent = (event: ChangeEvent<HTMLInputElement>) => {
			const dateString = event.currentTarget.value;
			if (dateString) {
				const parsedDate = parseDateString(dateString);
				setSelectedDate(parsedDate);
			} else {
				setSelectedDate(null);
			}
			if (typeof onChange === 'function') {
				onChange(dateString);
			}
		};

		const indicator = isLoading ? (
			<ProgressSpinner size={size} />
		) : (
			<Icon icon={icon} size={size} />
		);

		const label = valueLabel ? (
			<Text size={textSizeMap[size]}>{valueLabel}</Text>
		) : null;

		const handleCalendarChange = useCallback(
			(date: DateValue) => {
				setSelectedDate(date);
				if (typeof onChange === 'function') {
					onChange(formatDateToString(date));
				}
				// Close the popover after date selection
				if (popoverState) {
					popoverState.close();
				}
			},
			[onChange, popoverState],
		);

		const calendarProps: CalendarProps = useMemo(
			() => ({
				calendarOptions: {
					// For controlled components, use value. For uncontrolled, use defaultValue
					...(isControlled
						? {
								value:
									selectedDate || today(getLocalTimeZone()),
							}
						: {
								defaultValue:
									selectedDate || today(getLocalTimeZone()),
							}),
					...(min && { minValue: parseDateString(min) }),
					...(max && { maxValue: parseDateString(max) }),
				},
				...calendarOptions,
				onChange: handleCalendarChange,
			}),
			[
				selectedDate,
				calendarOptions,
				handleCalendarChange,
				isControlled,
				min,
				max,
			],
		);

		const contentCalendar = useMemo(
			() => <Calendar {...calendarProps} />,
			[calendarProps],
		);

		// Use native picker only if explicitly requested
		if (useNativePicker) {
			return (
				<div
					className={clsx(
						className,
						styles.inputContainer,
						disabled && styles.disabled.native,
					)}
				>
					<input
						{...inputProps}
						ref={ref}
						className={elementStyles({
							className: [
								styles.input,
								{
									[styles.disabled.cursor]: disabled,
								},
							],
						})}
						type="date"
						disabled={disabled}
						min={min}
						max={max}
						onChange={onChangeEvent}
					/>
					<div className={styles.inputOverlay}>
						<div
							className={clsx(styles.contents.default, {
								[styles.contents.withLabel]:
									Boolean(valueLabel),
							})}
						>
							{indicator}
							{label}
						</div>
					</div>
				</div>
			);
		}

		return (
			<div className={className}>
				<PopoverTrigger
					content={contentCalendar}
					placement={placement}
					testId={testId}
					isDisabled={disabled}
					onStateReady={setPopoverState}
				>
					<div
						className={sprinkles({
							display: 'flex',
							alignItems: 'center',
							gap: '1',
						})}
					>
						{indicator}
						{label}
					</div>
					<VisuallyHidden>
						{lang?.openCalendar ?? defaultEnglish.openCalendar}
					</VisuallyHidden>
				</PopoverTrigger>
				<input
					{...inputProps}
					ref={ref}
					value={formatDateToString(selectedDate)}
					type="hidden"
				/>
			</div>
		);
	},
);

DatePicker.displayName = 'DatePicker';
