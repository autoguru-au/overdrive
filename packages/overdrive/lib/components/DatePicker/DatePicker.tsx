import { CalendarIcon, IconType } from '@autoguru/icons';
import clsx from 'clsx';
import * as React from 'react';
import { ChangeEvent, ComponentProps, FunctionComponent } from 'react';

import { resolveResponsiveStyle } from '../../utils/resolveResponsiveProps';
import { Box, useBoxStyles } from '../Box';
import { Icon } from '../Icon';
import * as iconStyles from '../Icon/Icon.css';
import { ProgressSpinner } from '../ProgressSpinner';

import * as styles from './DatePicker.css';

export interface Props
	extends Partial<Pick<HTMLInputElement, 'min' | 'max' | 'value'>>,
		Pick<ComponentProps<typeof Icon>, 'size'> {
	className?: string;
	disabled?: boolean;
	icon?: IconType;
	isLoading?: boolean;

	onChange(date: string);
}

export const DatePicker: FunctionComponent<Props> = ({
	className = '',
	icon = CalendarIcon,
	size = 'medium',
	disabled = false,
	isLoading = false,
	onChange,
	...inputProps
}) => {
	const onChangeEvent = (event: ChangeEvent<HTMLInputElement>) => {
		if (typeof onChange === 'function') {
			onChange(event.currentTarget.value);
		}
	};
	return (
		<Box
			position="relative"
			overflow="hidden"
			className={clsx(
				resolveResponsiveStyle(size, iconStyles.size),
				className,
				{
					[styles.disabled.default]: disabled,
					[styles.disabled.root]: disabled,
				},
			)}>
			<Box
				position="absolute"
				height="full"
				width="full"
				is="input"
				disabled={disabled}
				onChange={onChangeEvent}
				className={clsx(
					{
						[styles.disabled.default]: disabled,
					},
					styles.input,
				)}
				type="date"
				{...inputProps}
			/>
			{isLoading ? (
				<ProgressSpinner
					className={clsx(
						styles.spinner,
						useBoxStyles({
							position: 'absolute',
						}),
					)}
					size={
						size as ComponentProps<typeof ProgressSpinner>['size']
					}
				/>
			) : (
				<Icon icon={icon} size={size} />
			)}
		</Box>
	);
};
