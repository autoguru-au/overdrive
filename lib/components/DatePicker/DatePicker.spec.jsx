import { render } from '@testing-library/react';
import * as React from 'react';

import { DatePicker } from './DatePicker';

// Mock window.matchMedia since it's not available in test environment
Object.defineProperty(window, 'matchMedia', {
	writable: true,
	value: vi.fn().mockImplementation((query) => ({
		matches: false,
		media: query,
		onchange: null,
		addListener: vi.fn(),
		removeListener: vi.fn(),
		addEventListener: vi.fn(),
		removeEventListener: vi.fn(),
		dispatchEvent: vi.fn(),
	})),
});

const TestIcon = (
	<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
		<path d="M183.253 353.707L280.96 256l-97.707-97.92 30.08-30.08 128 128-128 128-30.08-30.293z" />
	</svg>
);

describe('<DatePicker />', () => {
	it('should not throw', () =>
		expect(() => render(<DatePicker onChange={() => {}} />)).not.toThrow());

	it('should match snapshot without label', () => {
		expect(
			render(<DatePicker onChange={() => {}} />).container.firstChild,
		).toMatchSnapshot();
	});
});
