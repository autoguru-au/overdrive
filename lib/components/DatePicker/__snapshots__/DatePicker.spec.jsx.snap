// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DatePicker /> > should match snapshot without label 1`] = `
<div
  class="reset_block__1oah0223 useBoxStyles_position_relative__3xgq1lg0 useBoxStyles_overflow_hidden__3xgq1lg4"
>
  <input
    class="reset_trimmedBlockElement__1oah0225 reset_appearance__1oah0220 reset_input__1oah0228 useBoxStyles_width_full__3xgq1lfv useBoxStyles_height_full__3xgq1lfx useBoxStyles_position_absolute__3xgq1lfy DatePicker_input__1j3uugj0"
    type="date"
  />
  <div
    class="reset_block__1oah0223 DatePicker_contents_default__1j3uugj1"
  >
    <span
      class="reset_inlineText__1oah0222 useBoxStyles_display_block__3xgq1lg9 Icon_makeResponsiveStyle_key_bp__1gy3ire4"
    >
      <svg
        aria-hidden="true"
        class="useBoxStyles_display_block__3xgq1lg9 useBoxStyles_width_full__3xgq1lfv useBoxStyles_height_full__3xgq1lfx"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 18.998H5v-11h14m-3-7v2H8v-2H6v2H5a1.99 1.99 0 00-1.99 2l-.01 14a2 2 0 002 2h14c1.102 0 2-.896 2-2v-14a2 2 0 00-2-2h-1v-2m-1 11h-5v5h5v-5z"
        />
      </svg>
    </span>
  </div>
</div>
`;
